blog/2022-03-30-SDL-airbyte/2022-03-30-SDL-airbyte.md

```diff
@@ -0,0 +1,169 @@
+* try the connector: 
+  ```Bash
+  cd SDL_airbyte
+  python source_github/main.py spec | python -m json.tool
+  ```
+  This provides a [JSON string](github_spec_out.json) with the connector specification. The fields listed under `properties` are relevant for the configuration (compare with the configuration later used). 
+
+## Configuration
+To launch Smart Data Lake Builder (SDLB) with the Airbyte connector the following needs to be modified:
+
+* add the Airbyte ***dataObject*** with its configuration to the `config/application.conf`:
+  ```Python
+  dataObjects {
+    ext-commits {
+      type = AirbyteDataObject
+      config = {
+        "credentials": {
+          "personal_access_token": "<yourPersonalAccessToken>" ### enter your personal access token here
+        },
+        "repository": "smart-data-lake/smart-data-lake",
+        "start_date": "2021-02-01T00:00:00Z",
+        "branch": "documentation develop-spark3 develop-spark2",
+        "page_size_for_large_streams": 100
+      },
+      streamName = "commits",
+      cmd = {
+        type = CmdScript
+        name = "airbyte_connector_github"
+        linuxCmd = "python3 /mnt/source-github/main.py"
+      }
+    }
+  ...
+    stg-commits {
+     type = DeltaLakeTableDataObject
+     path = "~{id}"
+     table {
+      db = "default"
+      name = "stg_commits"
+      primaryKey = [created_at]
+      }
+    }
+  ```
+  The `ext-commits` defined connector settings. 
+  While the `config` varies from connector to connector, the remaining fields are SDL specific. 
+  The `streamName` selects the stream, exactly one. 
+  If multiple streams should be collected, multiple dataObjects need to be defined. 
+  In `linuxCmd` the actual connector script is called. 
+  In our case we will mount the connector directory into the SDL container. 
+
+* also add the definition of the data stream ***action*** to pipe the coming data stream into a `DeltaLakeTableDataObject`:
+  ```Bash
+    actions {
+      download-commits {
+        type = CopyAction
+        inputId = ext-commits
+        outputId = stg-commits
+        metadata {
+          feed = download
+        }
+      }
+  ...
+  ```
+* Since Airbyte will be called as Python script in the sdl container, we need to (re-)build the container with Python support and the Python `airbyte-cdk` package. 
+  Therefore, in the Dockerfile we add:
+	```
+	RUN \
+  apt update && \
+  apt --assume-yes install python3 python3-pip && \
+  pip3 install airbyte-cdk~=0.1.25
+  ```
+  and rebuild 
+  ```Bash
+  podman build . -t sdl-spark
+  ```
+
+Now we are ready to go. My full [SDLB config file](application.conf) additionally includes the pull-request stream.
+
+## Run and inspect results
+Since the data will be streamed into a `DeltaLakeTableDataObject` the metastore container is necessary. Further we aim to inspect the data using the Polynote notebook. Thus, first these containers are launched using (in the SDL example base directory):
```

Comment:
```suggestion
Since the data will be streamed into a `DeltaLakeTableDataObject`, the metastore container is necessary. Further, we aim to inspect the data using the Polynote notebook. Thus, first these containers are launched using (in the SDL example base directory):
```